src/localplayer.cpp

```diff
@@ -170,8 +170,8 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 		std::vector<CollisionInfo> *collision_info)
 {
 	if (!collision_info || collision_info->empty()) {
-		// Node below the feet, update each ClientEnvironment::step()
-		m_standing_node = floatToInt(m_position, BS) - v3s16(0, 1, 0);
+		// Node at feet position, update each ClientEnvironment::step()
+		m_standing_node = floatToInt(m_position, BS);
```

Comment:
Why change this, was it a bug?

Reply:
Why is it obviously wrong? The player's feet are now at `0,0,0` and this code is relevant when there are no collisions (falling down, liquids). Imagine the player is floating on a 1/16th slab (`free_move` enabled) - the current code will return the feet position (slab node) instead of the one below (dirt). This was _obviously wrong_ before.

Reply:
```
	/*
		Check properties of the node on which the player is standing
	*/
	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(m_standing_node));
	// Determine if jumping is possible
```
Yeah the point should be slightly below feet position, somewhere in code (lighting probably) we use a tiny offset to move the point below the edge of a node.

Reply:
According to the collision detection code, distances up to 0.15m away from the next colliding face is considered as "ground".
https://github.com/minetest/minetest/blob/622e2235ee700cf92843d2e99a47e36fd4508cf0/src/collision.cpp#L581

Subtracting any value from the position will cause wrong values for thin slabs. Nevertheless, this is only a default value which is overwritten as soon there is any collision. `m_standing_node` is only used *after* the collision detection and the standing node calculation.
Would you please be so nice and show me a location in the code where the standing node position is required to be exactly a bit below the player's feet? They could also stand on the bottom of a ladder, where the ground node doesn't matter at all.

Reply:
Ok thanks for the explanation, this is fine as it is then.